src/anemoi/inference/inputs/ekd.py

```diff
@@ -133,15 +117,24 @@ def _create_input_state(
         fields = input_state["fields"]
 
         input_fields = self._filter_and_sort(input_fields, variables=variables, dates=dates, title="Create input state")
-        mask = self.checkpoint.grid_points_mask
-        mask = ApplyMask(mask) if mask is not None else NoMask()
 
+        # target information
+        if mask is not None:
+            LOG.info("Applying mask from supporting arrays: %s", mask)
+            mask = self.checkpoint.load_supporting_array(mask)
```

Comment:
Could you raise a meaningful error here if this fails? Something like `"An input mask {mask} was supplied but the supporting array was not found in the checkpoint"` or something like that.

Reply:
There are already meaningful messages further down: https://github.com/ecmwf/anemoi-inference/blob/cc7fed54aa5f1923da7ef08649b102e2e690a3bf/src/anemoi/inference/metadata.py#L729 

is this enough or should I add something here as well?

Reply:
Ah yes, that is enough.